### PR TITLE
feat: Compute flat correlation.

### PR DESCRIPTION
**Pull request by riga**
This PR adds a function for computing the correlation matrix. The return value is a flat, square matrix.

Should there also be (or should the only way be) a way to return a version of the params pytree with parameter combinatorics?

### CODE REVIEW

File: src/evermore/loss.py [hunk 1]
```diff
@@ -66,3 +72,21 @@ def _constraint(param: Parameter) -> Array:
 
     # constraints from pdfs
     return _params_map(_constraint, params)
+
+
+def compute_correlation(
+    loss_fn: collections.abc.Callable,
```
Comment (pfackeldey) [inline src/evermore/loss.py:78]: `collections.abc.Callable` -> `tp.Callable` ? (and then let's drop the collections import)

File: src/evermore/loss.py [hunk 2]
```diff
@@ -66,3 +72,21 @@ def _constraint(param: Parameter) -> Array:
...
+    params: PyTree,
+    args: tuple[tp.Any, ...] = (),
```
Comment (pfackeldey) [inline src/evermore/loss.py:80]: do we want to pass `kwargs` through as well?

File: src/evermore/loss.py [hunk 3]
```diff
@@ -66,3 +72,21 @@ def _constraint(param: Parameter) -> Array:
...
+) -> Array:
+    # create a flattened version of the parameters and the loss
```
Comment (pfackeldey) [inline src/evermore/loss.py:82]: could you add a doc string with a simple example? (and do we need a test?)

**Review by pfackeldey — CHANGES_REQUESTED** [review]

**Comment by pfackeldey** [comment]
> Should there also be (or should the only way be) a way to return a version of the params pytree with parameter combinatorics?

I think it's good that it's a square matrix :+1:

File: src/evermore/loss.py (hunk 3 above)
Comment (riga) [reply src/evermore/loss.py:116]: I added both.

**Comment by riga** [comment]
Fyi, looking into other libraries (e.g. combine), it made more sense to call it `compute_covariance`, so I renamed it. [jax.random.multivariate_normal](https://docs.jax.dev/en/latest/_autosummary/jax.random.multivariate_normal.html) also accepts an argument `cov`, so this might be more consistent.

**Review by pfackeldey — APPROVED** [review]
Awesome 🎉 looks great to me!
